safe_transaction_service/history/signals.py

```diff
@@ -112,6 +115,36 @@ def safe_master_copy_clear_cache(
     SafeMasterCopy.objects.get_version_for_address.cache_clear()
 
 
+def get_safe_addresses_from_instance(
+    instance: Union[
+        TokenTransfer,
+        InternalTx,
+        MultisigConfirmation,
+        MultisigTransaction,
+    ]
+) -> Tuple[Optional[ChecksumAddress], Optional[ChecksumAddress]]:
```

Comment:
Document this method, not easy to understand what it does. Also why a tuple?  You are taking for granted that at most it will be 2 Safes affected, in the future there could be a database model having more (e.g. a 1155 multitransfer)